src/hm/hmRadio.h

```diff
@@ -105,61 +105,88 @@ class HmRadio : public Radio {
         }
 
         void loop(void) {
-            if (!mIrqRcvd)
-                return; // nothing to do
-            mIrqRcvd = false;
-            bool tx_ok, tx_fail, rx_ready;
-            mNrf24->whatHappened(tx_ok, tx_fail, rx_ready);  // resets the IRQ pin to HIGH
-            mNrf24->flush_tx();                              // empty TX FIFO
-
-            // start listening
-            uint8_t chOffset = 2;
-            mRxChIdx = (mTxChIdx + chOffset) % RF_CHANNELS;
-            mNrf24->setChannel(mRfChLst[mRxChIdx]);
-            mNrf24->startListening();
+            if (!mIrqRcvd && !mNRFisInRX)
+                return;         // first quick check => nothing to do at all here
 
             if(NULL == mLastIv) // prevent reading on NULL object!
                 return;
 
-            uint32_t innerLoopTimeout = 55000;
-            uint32_t loopMillis       = millis();
-            uint32_t outerLoopTimeout = (mLastIv->mIsSingleframeReq) ? 100 : ((mLastIv->mCmd != AlarmData) && (mLastIv->mCmd != GridOnProFilePara)) ? 400 : 600;
-            bool isRxInit             = true;
+            if(!mIrqRcvd) {     // no news from nRF, check timers
+                if (mRadioWaitTime.isTimeout()) { // timeout reached!
+                    mNRFisInRX = false;
+                    // add stop listening?
```

Comment:
Eigentlich könnten wir den NRF ja im default / immer im RX mode laufen lassen, das ist dann schon fast wie der Promiscuous Mode.
Nur wenn wir eine Anfrage senden wollen, müssen wir in den TX mode wechseln.

Reply:
Weiß nicht: Die eingehenden Nachrichten müssen ja irgendwie einem Inverter zugeordnet werden, und das macht der rx-Teil im Moment nicht (mehr? war in den alten Fassungen evtl. anders). 
Und auch die Kanalwechsel müßte man organisieren. Nach welcher Vorgabe?